### PR TITLE
fixing frontend issues

### PR DESCRIPTION
**Pull request by mjamei19**

### CODE REVIEW

**Comment by ccoffrin** [comment]
Thanks!  Should tests be added so that we don't have regressions on the FE?

**Comment by ccoffrin** [comment]
@eperim, @nicoleepp, I am ready to merge this, objections?

File: src/frontend/frontend.jl [hunk 1]
```diff
@@ -896,27 +895,21 @@ function add_cost_gen!(
             warn(LOGGER, w * "Using id = $element_id instead.")
         end
     end
-    if ismissing(gen_id)
-        push!(net.cost_gen, Any[coeffs, element_id, model, ncost])
+    push!(net.cost_gen, Any[coeffs, element_id, model, ncost])
```
Comment (eperim) [inline src/frontend/frontend.jl]: This push should only happen if `gen_id in gen_ids`, so this line should be moved just before line 901.

File: src/frontend/frontend.jl [hunk 2]
```diff
@@ -942,31 +935,22 @@ function add_cost_load!(
             warn(LOGGER, w * "Using id = $element_id instead.")
         end
     end
-    if ismissing(load_id)
-        push!(net.cost_load, Any[coeffs, element_id, model, ncost])
+    push!(net.cost_load, Any[coeffs, element_id, model, ncost])
```
Comment (eperim) [inline src/frontend/frontend.jl]: same thing here, this should go inside the `if`

File: src/frontend/frontend.jl [hunk 3]
```diff
@@ -992,27 +976,21 @@ function add_cost_gen!(
             warn(LOGGER, w * "Using id = $element_id instead.")
         end
     end
-    if ismissing(gen_id)
-        push!(net.cost_gen, Any[pwl_cost, element_id, model, ncost])
+    push!(net.cost_gen, Any[pwl_cost, element_id, model, ncost])
```
Comment (eperim) [inline src/frontend/frontend.jl]: This also should be moved inside the `if` below.

File: src/frontend/frontend.jl [hunk 4]
```diff
@@ -1038,19 +1016,12 @@ function add_cost_load!(
             warn(LOGGER, w * "Using id = $element_id instead.")
         end
     end
-    if ismissing(load_id)
-        push!(net.cost_load, Any[pwl_cost, element_id, model, ncost])
+    push!(net.cost_load, Any[pwl_cost, element_id, model, ncost])
```
Comment (eperim) [inline src/frontend/frontend.jl]: This also should be moved inside the if below.

File: test/frontend/frontend.jl [hunk 5]
```diff
@@ -85,32 +86,13 @@ end
...
-    @test size(PMA.lines(net))[1] == 21
-    @test size(PMA.buses(net))[1] == 15
-    @test size(PMA.generators(net))[1] == 6
-    @test size(PMA.loads(net)["ps_load"])[1] == 1
-    @test size(PMA.loads(net)["pi_load"])[1] == 13
-    build_pmc!(net)
-    @test length(PMA.pmc(net)["bus"]) == 15
-    old_rate = PMA.pmc(net)["branch"]["1"]["rate_a"]
-    PMA.max_load_percent!(PMA.pmc(net), 50)
-    @test PMA.pmc(net)["branch"]["1"]["rate_a"] == 0.5 * old_rate
```
Comment (eperim) [inline test/frontend/frontend.jl:113]: Is there a reason for removing these tests?

**Review by eperim — CHANGES_REQUESTED** [review]

File: src/frontend/frontend.jl (hunk 1 above)
Comment (mjamei19) [reply src/frontend/frontend.jl]: good catch! I made the changes in the new commit.

File: test/frontend/frontend.jl (hunk 5 above)
Comment (mjamei19) [reply test/frontend/frontend.jl:113]: These tests seemed repetitive to me. We have all of them in both `Piecewise linear costs` and `Polynomial costs`.  I kept those we did not have before but I had missed the `infeasible` test as it is added back. Pleas let me know if there is more thing to be added back in this test.

**Review by eperim — APPROVED** [review]

**Pull request merged**